src/components/MovieModal/MovieModal.tsx

```diff
@@ -35,35 +50,105 @@ export const genreOptions: readonly GenreOption[] = [
 ];
 
 const MovieModal = () => {
-  const { modal, editedPost } = useContext(AppContext);
+  const { modal, editedMovie, toggleModalType } = useContext(AppContext);
   const animatedComponents = makeAnimated();
   const isEditedPost = modal === ModalValue.EDIT;
-
   let movieDate: Date;
```

Comment:
Avoid this type of variable declarations, better use ternary operator instead
`const movieDate = isEditedPost ? new Date(editedMovie.release_date) : some default value`